src/net/net.c

```diff
@@ -83,7 +84,11 @@ int net_dst_source_addr_get(struct sa *dst, struct sa *ip)
 	if (err)
 		return err;
 
-	err = udp_connect(us, dst);
+	dsttmp = *dst;
+	if (!sa_port(&dsttmp))
+		sa_set_port(&dsttmp, 53);
```

Comment:
I dont think we should hardcode port 53 (DNS) into the library

Reply:
This is only a dummy connection, so no real traffic. Its only for setting up the socket. macOS refuses port 0 as destination port.

Reply:
what if the routing table has different rules for port 53 and port 5060 ?

it would make sense to use the same port number as the application would use.

Reply:
Good point, but its only a fallback if the application does not provide a valid dst port. So maybe returning a error code for dst port 0 is a better option and let the application handle this case.

Reply:
ok. Then I drop this and will handle this in retest. And we have to solve this also in ua.c.